examples/gateway/mlflow_serving/README.md

```diff
@@ -0,0 +1,181 @@
+the `sentence_transformers` library and the corresponding MLflow flavor. 
+
+```python
+from sentence_transformers import SentenceTransformer
+import mlflow
+
+
+model = SentenceTransformer(model_name_or_path="all-MiniLM-L6-v2")
+artifact_path = "embeddings_model"
+
+with mlflow.start_run():
+    mlflow.sentence_transformers.log_model(
+        model,
+        artifact_path=artifact_path,
+    )
+    model_uri = mlflow.get_artifact_uri(artifact_path)
+
+# Generate the cli command for starting a local MLflow Model Serving endpoint for this embeddings model
+print(f"mlflow models serve -m {model_uri} -h 127.0.0.1 -p 9020 --no-conda")
+```
+
+Copy the output from the print statement to the clipboard.
+
+## Starting the model server for the embeddings model
+With the printed string from running the above command copied to the clipboard, open a new terminal 
+and paste the string. Leave the terminal window open and running. 
+
+```commandline
+mlflow models serve -m file:///Users/me/demos/mlruns/0/2bfcdcb66eaf4c88abe8e0c7bcab639e/artifacts/embeddings_model -h 127.0.0.1 -p 9020 --no-conda
+```
+
+## Update the config.yaml to add a new embeddings route
+After assigning a valid port and ensuring that the model server starts correctly:
+
+```commandline
+2023/08/08 17:36:44 INFO mlflow.models.flavor_backend_registry: Selected backend for flavor 'python_function'
+2023/08/08 17:36:44 INFO mlflow.pyfunc.backend: === Running command 'exec gunicorn --timeout=60 -b 127.0.0.1:9020 -w 1 ${GUNICORN_CMD_ARGS} -- mlflow.pyfunc.scoring_server.wsgi:app'
+[2023-08-08 17:36:45 -0400] [54917] [INFO] Starting gunicorn 20.1.0
+[2023-08-08 17:36:45 -0400] [54917] [INFO] Listening at: http://127.0.0.1:9020 (54917)
+[2023-08-08 17:36:45 -0400] [54917] [INFO] Using worker: sync
+[2023-08-08 17:36:45 -0400] [54919] [INFO] Booting worker with pid: 54919
+```
+
+The flask app is ready to receive traffic. 
+
+Update the MLflow AI Gateway configuration file (config.yaml) with the new route:
+
+```yaml
+routes:
+  - name: embeddings
+    route_type: llm/v1/embeddings
+    model:
+      provider: mlflow-model-serving
+      name: sentence-transformer
+      config:
+        model_server_url: http://127.0.0.1:9020
+```
+
+The key component here is the ``model_server_url``. For serving an MLflow LLM, this url must match to the service that you are specifying for the 
+Model Serving server. 
+
+> **NOTE:** The MLflow Model Server does not have to be running in order to update the configuration file or to start the MLflow AI Gateway. In order to respond to submitted queries, it is required to be running.
+
+
+## Creating and logging a fill mask model
+To support an additional route for generating a mask fill response from masked input text, we need to log an appropriate model. 
+For this tutorial example, we'll use a `transformers` `Pipeline` wrapping a `BertForMaskedLM` torch model and will log this pipeline using the MLflow `transformers` flavor.
```

Comment:
Is this a super common thing that users would try to do with a completions endpoint? Is there a more canonical example? For example, I think it'd be better to show MPT-7B. We have that example for the Databricks AI Gateway docs (See go/aigateway).

Reply:
Added MPT-7B as an optional additional example (with caveats about not trying to run the example portion with this unless someone has a GPU instance that can handle the model). The fillmask will run just fine on CPU, which is why I chose it for the example.